### ISSUE TITLE
[Feature Request] Toggle dark mode for sites on global dark list

### ISSUE THREAD

**Issue by MPeti1** [darkreader/darkreader#4659]
**Is your feature request related to a problem? Please describe.**
There are sites where the dark mode is broken, or turned on by a script. 
For users who block scripts (including 1st party ones) the dark mode is not available.

**Describe the solution you'd like**
I think it would be suitable if the toggle button wouldn't be disabled for sites on the global dark list.

**Additional context**
In this screenshot you can see an example: on docs.docker.com toggling the dark mode toggle doesn't change anything, because 1st party scripts are blocked by default in by browser.
![image](https://user-images.githubusercontent.com/9215491/103488809-23ae4980-4e10-11eb-8ec8-560ff5590a63.png)

**Comment by Gusted** [comment]
Hi @MPeti1!

The toggle button is not disabled even on a dark site. Also `docs.docker.com` has been removed from the list #4453.

Regards,
Gusted

**Comment by MPeti1** [comment]
Thank you for the quick reply!

It's weird then, because clicking the button (inside the plugin window) does the click animation, but does not toggle dark mode, and also toggling it works for docker.com, but even if I set docker.com to be dark (with the plugin window), docs.docker.com is still light and has the x icon in the button

**Comment by Gusted** [comment]
Hi @MPeti1!

This sounds like a issue that's related to the #147 and their many related other issues.

Regards,
Gusted

**Comment by MPeti1** [comment]
Yes that one came to my mind too, but it seems to be different to me. Here the problem is that the subdomain does not follow the "main" domain's setting, so even if I turn it on on the main domain (www.docker.com), it won't be turned on on the subdomain (docs.docker.com)

**Comment by Gusted** [comment]
> Yes that one came to my mind too, but it seems to be different to me. Here the problem is that the subdomain does not follow the "main" domain's setting, so even if I turn it on on the main domain ([www.docker.com](http://www.docker.com)), it won't be turned on on the subdomain (docs.docker.com)

It might get conflicted with the dark-sites list, but I'm not sure.

**Comment by esotericist** [comment]
to add to the above, sometimes i have a site whose attempt at a "dark" theme is difficult to read for whatever reason, and i often can get better results at using the site's light theme and using Dark Reader to get what i want.

i'd much rather be able to do that than have to go to the lengths of having to manually twiddle css for a specific page in a style extension. isn't that what this extension plugin is for avoiding?

the end result is that the 'global dark list' has become, in a lot of cases, a "this site can't be made readable without an unreasonable amount of effort" list for me.

**Comment by Gusted** [comment]
> i'd much rather be able to do that than have to go to the lengths of having to manually twiddle css for a specific page in a style extension. isn't that what this extension plugin is for avoiding?

Depends. In an bird's eye context We see an site, we gather all styles we modify specific property's from that style into our own style and poof the site has an modified color scheme. Thereby in certain case the styles are already dark(Because it's by default or the site detects the system prefers an dark color-scheme) the 'modifier' adds little to nothing to these colors and thus making Dark Reader useless over the dark scheme. So yes this extension is avoiding to manually twiddle the CSS to an dark-scheme. 





> the end result is that the 'global dark list' has become, in a lot of cases, a "this site can't be made readable without an unreasonable amount of effort" list for me.

Uhm, excuse me? No, it isn't the only question to get an site into this config is when the answer is yes when `Regardless of the envirement, does it show an dark-scheme on every page and does Dark Reader add little to nothing to this dark-scheme. ` Nothing more Nothing less. Thereby such sites with this site can't be made readable are getting fixed when reported into the `dynamic-theme-fixes.config` with most of the time ~15 lines of CSS and proper configuration.

Regards,
Gusted

**Comment by esotericist** [comment]
i somehow missed that gusted responded to me back in february, so i want to try again:

my position is that the dark reader extension does more than just "make pages dark", it also provides crucial adjusters for brightness and contrast that i often find very necessary *even if the site already has a dark theme*.

so i still contend that if i have a site that is *almost* readable to me, and could be made readable with minor tweaks to brightness and contrast, but the global dark list prevents me from doing so? then the global dark list is a direct negative impact on accessibility, and i end up needing to do tedious manual css things with another style plugin in order to fix the problem.

given that my specific accessibility needs are *my* specific needs, attempting to get those into some centralized list makes little sense; so my request is that we simply have the local option to ignore the global dark list, so i can do what i need for me.

**Comment by MPeti1** [comment]
the current behaviour of the global dark list is also very bad for sites that are only dark if you enable javascript. I often find sites that I can't make dark just because it was put on the global dark list, but it isn't actually dark..

**Comment by Gusted** [comment]
>  just _made_ dark _by scripts_

Hmm, I personally don't see such sites that very often and if they do, they come with some kind of framework that won't even render any site if you have javascript disabled.



>  it also provides crucial adjusters for brightness and contrast that i often find very necessary _even if the site already has a dark theme_.

Correct, however darkening a dark page is very bad, it _always_ turns out to end up some weird color scheme.


@esotericist

I hear you, I see your need, however to just tweak it a little bit will also darkening a dark page which I just said don't look good at all. Il look what I can do.

**Comment by MPeti1** [comment]
> Hmm, I personally don't see such sites that very often and if they do, they come with some kind of framework that won't even render any site if you have javascript disabled.

I didn't want to imply that such sites are very regular, but that when a site is on that list, it always work like that.
An example to this is docs.docker.com: it renders perfectly, but the dark theme only works with a switch

**Comment by Gusted** [comment]
> An example to this is docs.docker.com: it renders perfectly, but the dark theme only works with a switch

Hmh, but it isn't dark-by-default and will never come on this list and quite frankly when someone pointed this out that this was on the list(few months ago?) it was removed immediately.

**Comment by MPeti1** [comment]
I think that was me (in the first screenshot of the post it was on the list), and only brought that as an example because I don't have a better example in my mind now

**Comment by esotericist** [comment]
so at some point the 'toggle current site' button started working for me even on sites that are on the global dark list. i didn't see explicit mention of it either here or in the commits, but i basically got exactly what i wanted and it has greatly assisted with my eyestrain issues on sites i found too complicated to hand edit the css on.

i just hope that it isn't by accident or the result of a bug, so i thought i should bring it up to help keep it from going away.

and since i failed to respond to this last month:

> I hear you, I see your need, however to just tweak it a little bit will also darkening a dark page which I just said don't look good at all. Il look what I can do.

for me it's not about looking good, it's about health and pain management. aesthetics take a distant back seat to that, which is why i'm so happy that dark reader is (at least at this moment) behaving in exactly the way i need

**Comment by Gusted** [comment]
Btw as for this Issue, you can toggle dark mode on sites that are in the global dark list and we very very much don't want to remove that ability as you should always be able to override the provided lists.

@esotericist as in your last section you mentioned something that I actually have been thinking about lately I'm going to put it up into a new issue/discussion.
